apps/frontend/src/app/pages/profile/profile.component.ts

```diff
@@ -269,4 +287,52 @@ export class ProfileComponent implements OnInit {
           });
       });
   }
+
+  showFriendsModal() {
+    this.friendsModalVisible = true;
+  }
+
+  toggleFriend(user: User) {
+    if (this.friendsToAdd.includes(user.id)) {
+      this.friendsToAdd = this.friendsToAdd.filter((u) => u !== user.id);
+    } else {
+      this.friendsToAdd.push(user.id);
+    }
+  }
+
+  selectAllFriends() {
+    this.friendsToAdd = this.steamFriends.map((user) => user.id);
+  }
+
+  deselectAllFriends() {
+    this.friendsToAdd = [];
+  }
+
+  addFollows() {
+    this.localUserService.followUsers(this.friendsToAdd).subscribe({
+      next: () => {
+        this.friendsModalVisible = false;
+        this.steamFriends = this.steamFriends.filter(
+          (user) => !this.friendsToAdd.includes(user.id)
+        );
+        this.friendsToAdd = [];
+
+        this.usersService.getUserFollows(this.user).subscribe({
+          next: (response) => (this.followingUsers = response.data),
+          error: (httpError: HttpErrorResponse) =>
+            this.messageService.add({
+              severity: 'error',
+              summary: 'Could not retrieve user follows',
+              detail: httpError.error.message
+            })
+        });
+      },
+      error: (httpError: HttpErrorResponse) =>
+        this.messageService.add({
+          severity: 'error',
+          summary: 'Could not add friends',
+          detail: httpError.error.message
+        })
+    });
+  }
```

Comment:
Again with naming, I think it's okay to keep this code in this class but given how large the profile component may get we should use "steamFriends" not "friends" and similar